File: src/main/java/radicalbit/io/QuickstartFirstJob.java
package radicalbit.io;

import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;
import java.util.List;

public class QuickstartFirstJob {

    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env =
                StreamExecutionEnvironment.getExecutionEnvironment();

        List<String> words =
                Arrays.asList("welcome", "to", "the", "fast-data", "revolution.");

        DataStream<String> source = env.fromCollection(words);

        DataStream<String> processed = source.map(new MapFunction<String, String>() {

            @Override
            public String map(String s) throws Exception {
                return s.toUpperCase();

            }
        });

        processed.print();

        env.execute("My First Job");
    }
}

File: src/main/scala/radicalbit/io/scala/QuickstartFirstJob.scala
package radicalbit.io.scala

import org.apache.flink.streaming.api.scala._

object QuickstartFirstJob {

  def main(args: Array[String]) {

    //Here we create the Streaming Execution environment...
    val env = StreamExecutionEnvironment.getExecutionEnvironment

    //...and we set its parallelism to 1 to see the results printed in order
    env.setParallelism(1)

    //We first create a DataStream from a bunch of strings...
    val source = env.fromElements("welcome", "to", "the", "fast-data", "revolution.")

    //...and we turn them to uppercase.
    val processed = source.map(_.toUpperCase)

    //At last we print the result to console.
    processed.print()

    env.execute("My first Job")
  }
}